builtin/update-index.c

```diff
@@ -347,7 +349,7 @@ static int process_directory(const char *path, int len, struct stat *st)
 	while (pos < active_nr) {
 		const struct cache_entry *ce = active_cache[pos++];
 
-		if (strncmp(ce->name, path, len))
+		if (!S_ISSPARSEDIR(ce->ce_mode) && strncmp(ce->name, path, len))
```

Comment:
Since this has a sequence of `if (condition) break;`, then maybe we should add another instead of using `&&` here.

Reply:
I don't think that would accomplish the same purpose (unless I'm misinterpreting). This condition appears to have been meant to identify that the current index entry is the "right" one to modify - previously that only required a name match, but this change adds that the entry must _also_ not be a sparse entry. I don't think I can/should separate the `!S_ISSPARSEDIR(...)` check into its own block (e.g., with a "continue") because it should continue with the other checks (potentially reaching the `error(...)`).

Reply:
You're right. I'm reading it wrong (it would work it if it was an `||`). Sorry.